### PR TITLE
feat(path): add replace_ext

### PR DESCRIPTION
**Pull request by illusory0x0**

### CODE REVIEW

File: internal/path/path_test.mbt [hunk 1]
```diff
@@ -139,3 +139,13 @@ test "ext/stem:edge cases" {
...
 }
+
+///|
+test "replace_ext" {
+  let path = "main.mbt"
+  inspect(replace_ext(path, ".c"), content="main.c")
+  let path = "main"
+  inspect(replace_ext(path, ".mbt"), content="main.mbt")
+  let path = "main.com/file"
+  inspect(replace_ext(path, ".c"), content="main.com/file.c")
```
Comment (tonyfettes) [inline internal/path/path_test.mbt]: Some test cases:

```moonbit
inspect(replace_ext("a.b.c.d", ".e"))
inspect(replace_ext("a", ".c"))
inspect(replace_ext("", ".c"))
inspect(replace_ext("/", ".c"))
inspect(replace_ext(".", ".c"))
inspect(replace_ext("..", ".c"))
inspect(replace_ext(".a.b", ".c"))
```

Comment (illusory0x0) [reply internal/path/path_test.mbt]: the `replace_ext` API designer cannot found the well defined behavior for some cases. If any `replace_ext` API user found well defined behavior for some case list in "replace_ext: undefined behavior". we can create a issue to discuss.
